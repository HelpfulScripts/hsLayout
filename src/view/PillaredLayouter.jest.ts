import * as hslayout from "../";
import m from "mithril";

let cont: any;
const root = window.document.createElement("div");

const layout = (type: string, contSizes: string[]) => {
        const config = {
            css: "myRow",
            content: [`First: ${contSizes[0]}`, "Second: remainder"]
        };
        config[type] = contSizes;
        m.mount(root, { view: () => m(hslayout.Layout, config) });
        cont = root.childNodes[0];
    }

describe("rows px", () => {
    const contSizes = ['150px','fill'];
    beforeAll(() => layout('rows', contSizes));
    describe("snapshots", () => {
        test("rows should be defined", () => expect(cont).toBeDefined());
        test(`hs-row-layout, [${contSizes}]`, () => expect(cont).toMatchSnapshot());
    });
});

describe("rows %", () => {
    const contSizes = ['30%','fill'];
    beforeAll(() => layout('rows', contSizes));
    describe("snapshots", () => {
        test("rows should be defined", () => expect(cont).toBeDefined());
        test(`hs-row-layout, [${contSizes}]`, () => expect(cont).toMatchSnapshot());
    });
});

describe("columns px", () => {
    const contSizes = ['150px','fill'];
    beforeAll(() => layout('columns', contSizes));
    describe("snapshots", () => {
        test("columns should be defined", () => expect(cont).toBeDefined());
        test(`hs-column-layout, [${contSizes}]`, () => expect(cont).toMatchSnapshot());
    });
});

describe("columns %", () => {
    const contSizes = ['30%','fill'];
    beforeAll(() => layout('columns', contSizes));
    describe("snapshots", () => {
        test("columns should be defined", () => expect(cont).toBeDefined());
        test(`hs-column-layout, [${contSizes}]`, () =>
            expect(cont).toMatchSnapshot());
    });
});
