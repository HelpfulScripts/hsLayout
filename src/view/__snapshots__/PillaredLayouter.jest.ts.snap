// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`columns % snapshots hs-column-layout, [30%,fill] 1`] = `
<div
  class="hs-layout  hs-column-layout myRow"
>
  <div
    class="hs-layout  "
    style="height: 100%; left: 0%; right: 70%; width: auto;"
  >
    <div
      class="hs-leaf"
    >
      First: 30%
    </div>
  </div>
  <div
    class="hs-layout  "
    style="height: 100%; left: 30%; right: 0%; width: auto;"
  >
    <div
      class="hs-leaf"
    >
      Second: remainder
    </div>
  </div>
</div>
`;

exports[`columns px snapshots hs-column-layout, [150px,fill] 1`] = `
<div
  class="hs-layout  hs-column-layout myRow"
>
  <div
    class="hs-layout  "
    style="height: 100%; left: 0px; width: 150px;"
  >
    <div
      class="hs-leaf"
    >
      First: 150px
    </div>
  </div>
  <div
    class="hs-layout  "
    style="height: 100%; left: 150px; right: 0px; width: auto;"
  >
    <div
      class="hs-leaf"
    >
      Second: remainder
    </div>
  </div>
</div>
`;

exports[`rows % snapshots hs-row-layout, [30%,fill] 1`] = `
<div
  class="hs-layout  hs-row-layout myRow"
>
  <div
    class="hs-layout  "
    style="width: 100%; top: 0%; bottom: 70%; height: auto;"
  >
    <div
      class="hs-leaf"
    >
      First: 30%
    </div>
  </div>
  <div
    class="hs-layout  "
    style="width: 100%; top: 30%; bottom: 0%; height: auto;"
  >
    <div
      class="hs-leaf"
    >
      Second: remainder
    </div>
  </div>
</div>
`;

exports[`rows px snapshots hs-row-layout, [150px,fill] 1`] = `
<div
  class="hs-layout  hs-row-layout myRow"
>
  <div
    class="hs-layout  "
    style="width: 100%; top: 0px; height: 150px;"
  >
    <div
      class="hs-leaf"
    >
      First: 150px
    </div>
  </div>
  <div
    class="hs-layout  "
    style="width: 100%; top: 150px; bottom: 0px; height: auto;"
  >
    <div
      class="hs-leaf"
    >
      Second: remainder
    </div>
  </div>
</div>
`;
